src/main/scala/inox/solvers/z3/Z3Native.scala

```diff
@@ -249,7 +250,9 @@ trait Z3Native extends ADTManagers with Interruptible { self: AbstractSolver =>
       case Implies(l, r) => z3.mkImplies(rec(l), rec(r))
       case Not(Equals(l, r)) => z3.mkDistinct(rec(l), rec(r))
       case Not(e) => z3.mkNot(rec(e))
-      case IntLiteral(v) => z3.mkInt(v, typeToSort(Int32Type))
+      case bv @ BVLiteral(_, size) if size <= 32 => z3.mkInt(bv.toBigInt.toInt, typeToSort(BVType(size)))
+      // FIXME currently ScalaZ3 doesn't support integer larger than Int(32).
```

Comment:
This is currently a latent issue with ScalaZ3 so I'm not sure it's easily fixable... It's probably not high priority either.